ravibeta/pravega
# update hadoopVersion to latest, add test override to use latest

**Change log description**  
This touches only one file gradle.properties to update the version of pravega-hdfs.
The hadoopVersion is updated to 2.9.2 as described in the linked issue. However, the guava version had to be lowered slightly. This is not a mistake. All revisions upwards of 2.9.2 up to the latest have had this breaking change which is described by the issues [HDFS-752](https://issues.apache.org/jira/browse/HDFS-752) and [HDFS-7477](https://issues.apache.org/jira/browse/HDFS-7477) and they don't agree with guava. The links for these issues are mentioned here so that the patch associated with these issues can be viewed. They are the only break-in changes we have. These changes are applicable to all recent versions but 2.9.2 as described in the attached issue fixes all the security vulnerabilities cited in the GitHub issue [4874](https://github.com/pravega/pravega/issues/4874) linked to this request.  

**Purpose of the change**  
The purpose of this change is to remove the security vulnerabilities associated with the hadoop version. The Hadoop Version poses a bigger security threat than others to Pravega.

Fixes #4874.

**What the code does**  
This code just changes versions in gradle.properties

**How to verify it**  
This change was verified by travis build.

Testing:
```
docker stack services pravega
ID                  NAME                   MODE                REPLICAS            IMAGE                                                                          PORTS
127hedtw2lzp        pravega_hdfs           replicated          1/1                 pravega/hdfs:2.7.7                                                        
182fk0erog8d        pravega_segmentstore   replicated          1/1                 devops-repo.isus.emc.com:8116/rajamr2/pravega:0.8.0-2577.5d687b9-SNAPSHOT      *:12345->12345/tcp
lcdxxqpdx3h9        pravega_zookeeper      replicated          1/1                 zookeeper:3.5.4-beta                                                      
m9xy1i32zerv        pravega_bookkeeper     replicated          3/3                 devops-repo.isus.emc.com:8116/rajamr2/bookkeeper:0.8.0-2577.5d687b9-SNAPSHOT
y8vrw86ysu7y        pravega_controller     replicated          1/1                 devops-repo.isus.emc.com:8116/rajamr2/pravega:0.8.0-2577.5d687b9-SNAPSHOT      *:9090->9090/tcp,*:10080->10080/tcp

```
Note that the exception described above is being addressed with different jars where the newer one is different from the older one) and the guava blogs also calls out this issue:
https://mvnrepository.com/artifact/com.google.guava/listenablefuture

Also, the jar used in the main is 29.0
lglbg153:~/hdfs # docker exec -it pravega_segmentstore.1.vjmdnfyfjy31n8aoemokeiu70 '/bin/sh'
/opt/pravega # find . | grep jar$ | grep -i guava
./lib/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar
./lib/guava-29.0-jre.jar
